### PR TITLE
Rem 393

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
sa-payment-terms

### CODE REVIEW

**Comment by Ghinwa-Arab** [comment]
image

add asterisk for type combo

File: src/pages/sa-payment-terms/index.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      })
+
+      return { ...response, _startAt: _startAt }
+    } catch (error) {}
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    access,
```
Comment (Ghinwa-Arab) [inline src/pages/sa-payment-terms/index.js:37]: add here invalidate and remove one below

File: src/pages/sa-payment-terms/index.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      },
+      width: 700,
+      height: 470,
+      title: _labels.paymentTerms
+    })
+  }
+
+  const del = async obj => {
+    await postRequest({
+      extension: SaleRepository.PaymentTerms.del,
```
Comment (Ghinwa-Arab) [inline src/pages/sa-payment-terms/index.js:104]: add try catch

File: src/pages/sa-payment-terms/index.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { useWindow } from 'src/windows'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { ControlContext } from 'src/providers/ControlContext'
+import PaaymentTermsForms from './forms/PaaymentTermsForms'
+
+const PaaymentTerms = () => {
```
Comment (omarZrayka23) [inline src/pages/sa-payment-terms/index.js:16]: name is PaymentTerms

File: src/pages/sa-payment-terms/index.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate, useResourceQuery } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { useWindow } from 'src/windows'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { ControlContext } from 'src/providers/ControlContext'
+import PaaymentTermsForms from './forms/PaaymentTermsForms'
```
Comment (omarZrayka23) [inline src/pages/sa-payment-terms/index.js:14]: fixe name to paymentTermsForms

File: src/pages/sa-payment-terms/index.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    access,
+    paginationParameters,
+    refetch
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: SaleRepository.PaymentTerms.qry,
+    datasetId: ResourceIds.PaymentTerm
+  })
+
+  const invalidate = useInvalidate({
```
Comment (omarZrayka23) [inline src/pages/sa-payment-terms/index.js:46]: clear useInvalidate and use invalidate in useResourceQuery

File: src/pages/sa-payment-terms/forms/PaaymentTermsForms.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: SaleRepository.PaymentTerms.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
```
Comment (omarZrayka23) [inline src/pages/sa-payment-terms/forms/PaaymentTermsForms.js:30]: recordId: recordId || null

**Pull request merged**